sdk/servicebus/service-bus/src/core/managementClient.ts

```diff
@@ -1255,7 +1256,9 @@ export class ManagementClient extends LinkEntity<RequestResponseLink> {
     if (
       typeof filter !== "boolean" &&
       typeof filter !== "string" &&
-      !correlationProperties.some((validProperty) => filter.hasOwnProperty(validProperty))
+      !correlationProperties.some((validProperty) =>
+        Object.hasOwnProperty.call(filter, validProperty)
```

Comment:
How does this change help?
What is wrong with `filter.hasOwnProperty`? (filter could be undefined??)

Reply:
This rule was failing: https://eslint.org/docs/rules/no-prototype-builtins
I made this change before using Chris's guards, so I can change to use the guards instead.
Updated in https://github.com/Azure/azure-sdk-for-js/pull/13935/commits/c485646f7a1dbe1f03f75761fe94e39c5e89ad7e